sunbeam/commands/juju.py

```diff
@@ -408,6 +399,39 @@ def run(self) -> Result:
 
         :return:
         """
+        try:
+            if self.states:
+
+                """
+                # Workaround for https://bugs.launchpad.net/juju/+bug/1990797
+                import os
+                snap_real_home = os.environ.get('SNAP_REAL_HOME')
+                home = os.environ.get('HOME')
+                src = Path(snap_real_home) / '.local' / 'share' / 'juju'
+                dst = Path(home) / '.local' / 'share'
+                os.makedirs(dst, exist_ok=True)
+                dst = dst / 'juju'
+                try:
+                    os.symlink(src, dst)
+                except FileExistsError:
+                    pass
+                LOG.debug('Symlink src {src} to dst {dst}')
+                """
+                home = os.environ.get('SNAP_REAL_HOME')
+                os.environ['JUJU_DATA'] = f'{home}/.local/share/juju'
```

Comment:
I'm fairly positive this makes it clear that you're essentially remapping it for the ability to have the python juju client work. As such, you can disregard my previous comment (though I don't mind confirmation if you have it).

I think this is probably the lesser of two evils for the two approaches. With the former one (e.g. the symlink) we'd have to remember to clean up the link in the future.